pkg/certificate/providers/config.go

```diff
@@ -158,6 +169,21 @@ func (c *MRCProviderGenerator) getHashiVaultOSMCertificateManager(mrc *v1alpha2.
 	return vaultClient, id, nil
 }
 
+// getHashiVaultOSMToken returns the Hashi Vault token
```

Comment:
nit: the comment could convey more meaning, e.g. `.... returns the Hashi Vault Token from ...`